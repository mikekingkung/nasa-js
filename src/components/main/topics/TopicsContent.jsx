import React from 'react';
import Grid from '@material-ui/core/Grid';
import Typography from '@material-ui/core/Typography';
import { makeStyles } from '@material-ui/core/styles';

const TopicsContent = () => {
  const useStyles = makeStyles((theme) => ({
    landingpage: {
      height: "100%",
    },
    container: {
      height: "100%",
      width: '100%',
      padding: "10px"
     },
     missionknowledecontainer: {
       width: "100%",
       minHeight: "100%"
     },
     missioncontextcontainer: {
       display: "flex",
       flexDirection: "row",
       justifyContent: "space-between",
       height: "100%",
       width: "100%",
       flex: 1,
       backgroundColor: "black",
       color: "red",
       padding: "20",
       minwidth: "200"
       
      },
     recentknowledgecontainer: {
       display: "flex",
       flexDirection: "row",
       justifyContent: "space-between",
       height: "100%",
       width: "100%",
       flex: 1,
       padding: "10px",
       backgroundColor: "black",
       color: "red",
      },
      textfield: {
        border: "1px solid blue",
      },
      imageorvideo: {
        padding: "20",
        minwidth: "200"
      }
  }));

  const classes = useStyles();
    return (
      <div>
        <Grid container className="classes.container">
          <Grid item xs={12} className="classes.missioncontextcontainer">               
             <h2>Topics</h2>
          </Grid>
          <Grid item xs={12} className="classes.missioncontextcontainer">
             <a href = "https://www.nasa.gov/topics/humans-in-space" target="_blank" rel="noreferrer">Humans in space</a>
          </Grid>
          <Grid item xs={12} className="classes.missioncontextcontainer">
             <a href = "https://www.nasa.gov/topics/moon-to-mars" target="_blank" rel="noreferrer">Moon to Mars</a>
          </Grid>
          <Grid item xs={12} className="classes.missioncontextcontainer">
             <a href = "https://www.nasa.gov/topics/earth/index.html" target="_blank" rel="noreferrer">Earth</a>
          </Grid>
          <Grid item xs={12} className="classes.missioncontextcontainer">          
             <a href = "https://www.nasa.gov/topics/technology/index.html" target="_blank" rel="noreferrer">Technology</a>
          </Grid>
          <Grid item xs={12} className="classes.missioncontextcontainer">
             <a href = "https://www.nasa.gov/topics/aeronautics/index.html" target="_blank" rel="noreferrer">Aeronautics</a>
          </Grid>               
          <Grid item xs={12} className="classes.missioncontextcontainer">
             <a href = "nasa.gov/topics/solarsystem/index.html" target="_blank" rel="noreferrer">Solar System</a>
          </Grid>               
          <Grid item xs={12} className="classes.missioncontextcontainer">
             <a href = "https://www.nasa.gov/stem" target="_blank" rel="noreferrer">STEM</a>
          </Grid>
          <Grid item xs={12} className="classes.missioncontextcontainer">
             <a href = "https://www.nasa.gov/topics/benefits/index.html" target="_blank" rel="noreferrer">Benefits</a>
          </Grid>
          <Grid item xs={12} className="classes.missioncontextcontainer">
             <h2>Missions</h2>
          </Grid>
          <Grid item xs={12} className="classes.missioncontextcontainer">
             <a href = "https://www.nasa.gov/specials/artemis/" target="_blank" rel="noreferrer">Artemis</a>
          </Grid> 
          <Grid item xs={12} className="classes.missioncontextcontainer">
             <a href = "https://www.nasa.gov/exploration/commercial/crew/index.html" target="_blank" rel="noreferrer">Commercial Crew Program</a>
          </Grid> 
          <Grid item xs={12} className="classes.missioncontextcontainer">
             <a href = "https://www.nasa.gov/mission_pages/msl/index.html" target="_blank" rel="noreferrer">Curiosity Mars Rover</a>
          </Grid> 
          <Grid item xs={12} className="classes.missioncontextcontainer">
             <a href = "https://www.nasa.gov/mission_pages/hubble/main/index.html" target="_blank" rel="noreferrer">Hubble Space Telescope</a>
          </Grid> 
          <Grid item xs={12} className="classes.missioncontextcontainer">
            <a href = "https://www.nasa.gov/mission_pages/webb/main/index.html" target="_blank" rel="noreferrer">James Webb Space Telescope</a>
          </Grid> 
          <Grid item xs={12} className="classes.missioncontextcontainer">
            <a href = "https://www.nasa.gov/mission_pages/juno/main/index.html" target="_blank" rel="noreferrer">Juno: Mission Jupiter</a>
          </Grid> 
          <Grid item xs={12} className="classes.missioncontextcontainer">
             <a href = "https://www.nasa.gov/mission_pages/station/main/index.html" target="_blank" rel="noreferrer">International Space Station</a>
          </Grid> 
          <Grid item xs={12} className="classes.missioncontextcontainer">
             <a href = "https://www.nasa.gov/mission_pages/insight/main/index.html" target="_blank" rel="noreferrer">Insight Mars Lander</a>
          </Grid> 
          <Grid item xs={12} className="classes.missioncontextcontainer">
            <a href = "https://www.nasa.gov/perseverance" target="_blank" rel="noreferrer">Insight Mars Lander</a>
          </Grid> 
          <Grid item xs={12} className="classes.missioncontextcontainer">
            <a href = "https://www.nasa.gov/mission_pages/newhorizons/main/index.html" target="_blank" rel="noreferrer">New Horizons</a>
          </Grid> 
          <Grid item xs={12} className="classes.missioncontextcontainer">
            <a href = "https://www.nasa.gov/osiris-rex" target="_blank" rel="noreferrer">Osiris Rex</a>
          </Grid> 
          <Grid item xs={12} className="classes.missioncontextcontainer">
            <a href = "https://www.nasa.gov/content/goddard/parker-solar-probe" target="_blank" rel="noreferrer">Parker Solar Probe</a>
          </Grid> 
          <Grid item xs={12} className="classes.missioncontextcontainer">
            <a href = "https://www.nasa.gov/launchschedule/" target="_blank" rel="noreferrer">Launch Schedule</a>
          </Grid> 
          <Grid item xs={12} className="classes.missioncontextcontainer">
            <a href = "https://www.nasa.gov/missions" target="_blank" rel="noreferrer">Missions A-Z</a>
          </Grid> 
         </Grid>
      </div>
     );
};

















export default TopicsContent;  


