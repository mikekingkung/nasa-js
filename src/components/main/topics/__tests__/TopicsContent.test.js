import React from 'react';
import { render} from '@testing-library/react';
import TopicsContent from '../TopicsContent';

describe('TopicsContent Page component', () => {
    it('Correctly renders' ,() => {
        const wrapper = render(<TopicsContent/>);
        expect(wrapper).toMatchSnapshot();
    });
});