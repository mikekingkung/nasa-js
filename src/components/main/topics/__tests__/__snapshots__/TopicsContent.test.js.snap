// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopicsContent Page component Correctly renders 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          class="MuiGrid-root classes.container MuiGrid-container"
        >
          <div
            class="MuiGrid-root classes.missioncontextcontainer MuiGrid-item MuiGrid-grid-xs-12"
          >
            <h2>
              Topics
            </h2>
          </div>
          <div
            class="MuiGrid-root classes.missioncontextcontainer MuiGrid-item MuiGrid-grid-xs-12"
          >
            <a
              href="https://www.nasa.gov/topics/humans-in-space"
              rel="noreferrer"
              target="_blank"
            >
              Humans in space
            </a>
          </div>
          <div
            class="MuiGrid-root classes.missioncontextcontainer MuiGrid-item MuiGrid-grid-xs-12"
          >
            <a
              href="https://www.nasa.gov/topics/moon-to-mars"
              rel="noreferrer"
              target="_blank"
            >
              Moon to Mars
            </a>
          </div>
          <div
            class="MuiGrid-root classes.missioncontextcontainer MuiGrid-item MuiGrid-grid-xs-12"
          >
            <a
              href="https://www.nasa.gov/topics/earth/index.html"
              rel="noreferrer"
              target="_blank"
            >
              Earth
            </a>
          </div>
          <div
            class="MuiGrid-root classes.missioncontextcontainer MuiGrid-item MuiGrid-grid-xs-12"
          >
            <a
              href="https://www.nasa.gov/topics/technology/index.html"
              rel="noreferrer"
              target="_blank"
            >
              Technology
            </a>
          </div>
          <div
            class="MuiGrid-root classes.missioncontextcontainer MuiGrid-item MuiGrid-grid-xs-12"
          >
            <a
              href="https://www.nasa.gov/topics/aeronautics/index.html"
              rel="noreferrer"
              target="_blank"
            >
              Aeronautics
            </a>
          </div>
          <div
            class="MuiGrid-root classes.missioncontextcontainer MuiGrid-item MuiGrid-grid-xs-12"
          >
            <a
              href="nasa.gov/topics/solarsystem/index.html"
              rel="noreferrer"
              target="_blank"
            >
              Solar System
            </a>
          </div>
          <div
            class="MuiGrid-root classes.missioncontextcontainer MuiGrid-item MuiGrid-grid-xs-12"
          >
            <a
              href="https://www.nasa.gov/stem"
              rel="noreferrer"
              target="_blank"
            >
              STEM
            </a>
          </div>
          <div
            class="MuiGrid-root classes.missioncontextcontainer MuiGrid-item MuiGrid-grid-xs-12"
          >
            <a
              href="https://www.nasa.gov/topics/benefits/index.html"
              rel="noreferrer"
              target="_blank"
            >
              Benefits
            </a>
          </div>
          <div
            class="MuiGrid-root classes.missioncontextcontainer MuiGrid-item MuiGrid-grid-xs-12"
          >
            <h2>
              Missions
            </h2>
          </div>
          <div
            class="MuiGrid-root classes.missioncontextcontainer MuiGrid-item MuiGrid-grid-xs-12"
          >
            <a
              href="https://www.nasa.gov/specials/artemis/"
              rel="noreferrer"
              target="_blank"
            >
              Artemis
            </a>
          </div>
          <div
            class="MuiGrid-root classes.missioncontextcontainer MuiGrid-item MuiGrid-grid-xs-12"
          >
            <a
              href="https://www.nasa.gov/exploration/commercial/crew/index.html"
              rel="noreferrer"
              target="_blank"
            >
              Commercial Crew Program
            </a>
          </div>
          <div
            class="MuiGrid-root classes.missioncontextcontainer MuiGrid-item MuiGrid-grid-xs-12"
          >
            <a
              href="https://www.nasa.gov/mission_pages/msl/index.html"
              rel="noreferrer"
              target="_blank"
            >
              Curiosity Mars Rover
            </a>
          </div>
          <div
            class="MuiGrid-root classes.missioncontextcontainer MuiGrid-item MuiGrid-grid-xs-12"
          >
            <a
              href="https://www.nasa.gov/mission_pages/hubble/main/index.html"
              rel="noreferrer"
              target="_blank"
            >
              Hubble Space Telescope
            </a>
          </div>
          <div
            class="MuiGrid-root classes.missioncontextcontainer MuiGrid-item MuiGrid-grid-xs-12"
          >
            <a
              href="https://www.nasa.gov/mission_pages/webb/main/index.html"
              rel="noreferrer"
              target="_blank"
            >
              James Webb Space Telescope
            </a>
          </div>
          <div
            class="MuiGrid-root classes.missioncontextcontainer MuiGrid-item MuiGrid-grid-xs-12"
          >
            <a
              href="https://www.nasa.gov/mission_pages/juno/main/index.html"
              rel="noreferrer"
              target="_blank"
            >
              Juno: Mission Jupiter
            </a>
          </div>
          <div
            class="MuiGrid-root classes.missioncontextcontainer MuiGrid-item MuiGrid-grid-xs-12"
          >
            <a
              href="https://www.nasa.gov/mission_pages/station/main/index.html"
              rel="noreferrer"
              target="_blank"
            >
              International Space Station
            </a>
          </div>
          <div
            class="MuiGrid-root classes.missioncontextcontainer MuiGrid-item MuiGrid-grid-xs-12"
          >
            <a
              href="https://www.nasa.gov/mission_pages/insight/main/index.html"
              rel="noreferrer"
              target="_blank"
            >
              Insight Mars Lander
            </a>
          </div>
          <div
            class="MuiGrid-root classes.missioncontextcontainer MuiGrid-item MuiGrid-grid-xs-12"
          >
            <a
              href="https://www.nasa.gov/perseverance"
              rel="noreferrer"
              target="_blank"
            >
              Insight Mars Lander
            </a>
          </div>
          <div
            class="MuiGrid-root classes.missioncontextcontainer MuiGrid-item MuiGrid-grid-xs-12"
          >
            <a
              href="https://www.nasa.gov/mission_pages/newhorizons/main/index.html"
              rel="noreferrer"
              target="_blank"
            >
              New Horizons
            </a>
          </div>
          <div
            class="MuiGrid-root classes.missioncontextcontainer MuiGrid-item MuiGrid-grid-xs-12"
          >
            <a
              href="https://www.nasa.gov/osiris-rex"
              rel="noreferrer"
              target="_blank"
            >
              Osiris Rex
            </a>
          </div>
          <div
            class="MuiGrid-root classes.missioncontextcontainer MuiGrid-item MuiGrid-grid-xs-12"
          >
            <a
              href="https://www.nasa.gov/content/goddard/parker-solar-probe"
              rel="noreferrer"
              target="_blank"
            >
              Parker Solar Probe
            </a>
          </div>
          <div
            class="MuiGrid-root classes.missioncontextcontainer MuiGrid-item MuiGrid-grid-xs-12"
          >
            <a
              href="https://www.nasa.gov/launchschedule/"
              rel="noreferrer"
              target="_blank"
            >
              Launch Schedule
            </a>
          </div>
          <div
            class="MuiGrid-root classes.missioncontextcontainer MuiGrid-item MuiGrid-grid-xs-12"
          >
            <a
              href="https://www.nasa.gov/missions"
              rel="noreferrer"
              target="_blank"
            >
              Missions A-Z
            </a>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        class="MuiGrid-root classes.container MuiGrid-container"
      >
        <div
          class="MuiGrid-root classes.missioncontextcontainer MuiGrid-item MuiGrid-grid-xs-12"
        >
          <h2>
            Topics
          </h2>
        </div>
        <div
          class="MuiGrid-root classes.missioncontextcontainer MuiGrid-item MuiGrid-grid-xs-12"
        >
          <a
            href="https://www.nasa.gov/topics/humans-in-space"
            rel="noreferrer"
            target="_blank"
          >
            Humans in space
          </a>
        </div>
        <div
          class="MuiGrid-root classes.missioncontextcontainer MuiGrid-item MuiGrid-grid-xs-12"
        >
          <a
            href="https://www.nasa.gov/topics/moon-to-mars"
            rel="noreferrer"
            target="_blank"
          >
            Moon to Mars
          </a>
        </div>
        <div
          class="MuiGrid-root classes.missioncontextcontainer MuiGrid-item MuiGrid-grid-xs-12"
        >
          <a
            href="https://www.nasa.gov/topics/earth/index.html"
            rel="noreferrer"
            target="_blank"
          >
            Earth
          </a>
        </div>
        <div
          class="MuiGrid-root classes.missioncontextcontainer MuiGrid-item MuiGrid-grid-xs-12"
        >
          <a
            href="https://www.nasa.gov/topics/technology/index.html"
            rel="noreferrer"
            target="_blank"
          >
            Technology
          </a>
        </div>
        <div
          class="MuiGrid-root classes.missioncontextcontainer MuiGrid-item MuiGrid-grid-xs-12"
        >
          <a
            href="https://www.nasa.gov/topics/aeronautics/index.html"
            rel="noreferrer"
            target="_blank"
          >
            Aeronautics
          </a>
        </div>
        <div
          class="MuiGrid-root classes.missioncontextcontainer MuiGrid-item MuiGrid-grid-xs-12"
        >
          <a
            href="nasa.gov/topics/solarsystem/index.html"
            rel="noreferrer"
            target="_blank"
          >
            Solar System
          </a>
        </div>
        <div
          class="MuiGrid-root classes.missioncontextcontainer MuiGrid-item MuiGrid-grid-xs-12"
        >
          <a
            href="https://www.nasa.gov/stem"
            rel="noreferrer"
            target="_blank"
          >
            STEM
          </a>
        </div>
        <div
          class="MuiGrid-root classes.missioncontextcontainer MuiGrid-item MuiGrid-grid-xs-12"
        >
          <a
            href="https://www.nasa.gov/topics/benefits/index.html"
            rel="noreferrer"
            target="_blank"
          >
            Benefits
          </a>
        </div>
        <div
          class="MuiGrid-root classes.missioncontextcontainer MuiGrid-item MuiGrid-grid-xs-12"
        >
          <h2>
            Missions
          </h2>
        </div>
        <div
          class="MuiGrid-root classes.missioncontextcontainer MuiGrid-item MuiGrid-grid-xs-12"
        >
          <a
            href="https://www.nasa.gov/specials/artemis/"
            rel="noreferrer"
            target="_blank"
          >
            Artemis
          </a>
        </div>
        <div
          class="MuiGrid-root classes.missioncontextcontainer MuiGrid-item MuiGrid-grid-xs-12"
        >
          <a
            href="https://www.nasa.gov/exploration/commercial/crew/index.html"
            rel="noreferrer"
            target="_blank"
          >
            Commercial Crew Program
          </a>
        </div>
        <div
          class="MuiGrid-root classes.missioncontextcontainer MuiGrid-item MuiGrid-grid-xs-12"
        >
          <a
            href="https://www.nasa.gov/mission_pages/msl/index.html"
            rel="noreferrer"
            target="_blank"
          >
            Curiosity Mars Rover
          </a>
        </div>
        <div
          class="MuiGrid-root classes.missioncontextcontainer MuiGrid-item MuiGrid-grid-xs-12"
        >
          <a
            href="https://www.nasa.gov/mission_pages/hubble/main/index.html"
            rel="noreferrer"
            target="_blank"
          >
            Hubble Space Telescope
          </a>
        </div>
        <div
          class="MuiGrid-root classes.missioncontextcontainer MuiGrid-item MuiGrid-grid-xs-12"
        >
          <a
            href="https://www.nasa.gov/mission_pages/webb/main/index.html"
            rel="noreferrer"
            target="_blank"
          >
            James Webb Space Telescope
          </a>
        </div>
        <div
          class="MuiGrid-root classes.missioncontextcontainer MuiGrid-item MuiGrid-grid-xs-12"
        >
          <a
            href="https://www.nasa.gov/mission_pages/juno/main/index.html"
            rel="noreferrer"
            target="_blank"
          >
            Juno: Mission Jupiter
          </a>
        </div>
        <div
          class="MuiGrid-root classes.missioncontextcontainer MuiGrid-item MuiGrid-grid-xs-12"
        >
          <a
            href="https://www.nasa.gov/mission_pages/station/main/index.html"
            rel="noreferrer"
            target="_blank"
          >
            International Space Station
          </a>
        </div>
        <div
          class="MuiGrid-root classes.missioncontextcontainer MuiGrid-item MuiGrid-grid-xs-12"
        >
          <a
            href="https://www.nasa.gov/mission_pages/insight/main/index.html"
            rel="noreferrer"
            target="_blank"
          >
            Insight Mars Lander
          </a>
        </div>
        <div
          class="MuiGrid-root classes.missioncontextcontainer MuiGrid-item MuiGrid-grid-xs-12"
        >
          <a
            href="https://www.nasa.gov/perseverance"
            rel="noreferrer"
            target="_blank"
          >
            Insight Mars Lander
          </a>
        </div>
        <div
          class="MuiGrid-root classes.missioncontextcontainer MuiGrid-item MuiGrid-grid-xs-12"
        >
          <a
            href="https://www.nasa.gov/mission_pages/newhorizons/main/index.html"
            rel="noreferrer"
            target="_blank"
          >
            New Horizons
          </a>
        </div>
        <div
          class="MuiGrid-root classes.missioncontextcontainer MuiGrid-item MuiGrid-grid-xs-12"
        >
          <a
            href="https://www.nasa.gov/osiris-rex"
            rel="noreferrer"
            target="_blank"
          >
            Osiris Rex
          </a>
        </div>
        <div
          class="MuiGrid-root classes.missioncontextcontainer MuiGrid-item MuiGrid-grid-xs-12"
        >
          <a
            href="https://www.nasa.gov/content/goddard/parker-solar-probe"
            rel="noreferrer"
            target="_blank"
          >
            Parker Solar Probe
          </a>
        </div>
        <div
          class="MuiGrid-root classes.missioncontextcontainer MuiGrid-item MuiGrid-grid-xs-12"
        >
          <a
            href="https://www.nasa.gov/launchschedule/"
            rel="noreferrer"
            target="_blank"
          >
            Launch Schedule
          </a>
        </div>
        <div
          class="MuiGrid-root classes.missioncontextcontainer MuiGrid-item MuiGrid-grid-xs-12"
        >
          <a
            href="https://www.nasa.gov/missions"
            rel="noreferrer"
            target="_blank"
          >
            Missions A-Z
          </a>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
